### ISSUE TITLE
Question about the released implementation of AGTAO

### ISSUE THREAD

**Issue by InuHund** [TiezMind/AGT-unlearning#2]
### Question about the AGTAO implementation and the code used for the reported experiments

I am currently trying to reproduce the AGTAO results reported in the paper and have been comparing the paper's formulation with the released implementation.

I noticed several points in the current implementation that I would like to clarify.

#### 1. Gradient used for the AO term

In the paper, the AO regularizer is defined using the forget and retain gradients:

$$
g_f = \nabla_\theta L_f,\qquad
g_r = \nabla_\theta L_r.
$$

However, in `AGT.py`, the gradient stored as the objective/unlearning gradient appears to be computed from:

```python
current_unlearn_loss = loss_npo + loss_r
```

before calling `store_grads()`.

If I understand the implementation correctly, this means that the gradient used as \(g_f\) for AO contains both the forget loss and the retain loss, rather than the forget loss alone.

Could you clarify whether this is intentional? In particular, should \(g_f\) for the AO term be the gradient of `loss_npo` only?

#### 2. Detachment of the stored gradients

In `store_grads()`, the gradients are stored using:

```python
target_dict[name] += param.grad.detach()
```

Since the AO regularizer is subsequently constructed from these stored gradients, it appears that the stored gradients are detached from the computation graph.

If so, my understanding is that the AO regularizer would not contribute a gradient with respect to the model parameters during the final backward pass.

Is this intentional? Or was the implementation used for the experiments different in this respect?

#### 3. Gamma value

Appendix A.3 of the paper reports:

```text
gamma = 1
```

and the configuration/README also appears to specify `ao_gamma: 1.0`.

However, the current implementation appears to pass `gamma=2.0` directly when computing the AO regularizer.

Could you clarify which value was used for the experiments reported in the paper?

---

More generally, were the results reported in the paper generated using the current public version of this repository?

If a different version of the implementation was used for the reported experiments, could you clarify which version or commit was used?

I may be misunderstanding some part of the implementation, so I would appreciate any clarification. Thank you!
